api/src/main/java/io/minio/MinioClient.java

```diff
@@ -927,6 +937,23 @@ private HttpResponse execute(Method method, String region, String bucketName, St
       this.traceStream.println(headers);
     }
 
+    SSLContext context = SSLContext.getInstance("TLSv1.2");
+
+    try {
+      context.init(null, null, null);
+    } catch (KeyManagementException e) {
+      throw new InternalException("SSL Context initialization failed.");
+    }
+
+    SSLSocketFactory socketFac = context.getSocketFactory();
+
+    ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+        .tlsVersions(TlsVersion.TLS_1_2)
+        .build();
+
+    this.httpClient.setConnectionSpecs(Collections.singletonList(spec));
+    this.httpClient.setSslSocketFactory(socketFac);
```

Comment:
* You could make this as a separate public method `public void enableTLSv12()`.  This make a choice to consumer to choose what they want than forcing to TLSv1.2.
* Its not required for each `execute()`
* you can directly like `this.httpClient.setSslSocketFactory(context.getSocketFactory());`